src/components/UI/navbar/index.js

```diff
@@ -1,50 +1,231 @@
+import Image from 'next/image';
+import {
+  PATHS,
+  LOGIN_URL,
+  USER_DATA_URL,
+  USER_PROFILE_URL,
+} from '@constants/AppConstants';
+import Link from 'next/link';
 
-const Navbar = ({ drawerToggleClicked }) => {
+import styles from './navbar.module.scss';
+
+const GenericClosePopUp = (ref, callback) => {
+  const handleClick = (e) => {
+    if (ref.current && !ref.current.contains(e.target)) {
+      callback();
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('click', handleClick);
+
+    return () => {
+      document.removeEventListener('click', handleClick);
+    };
+  });
+};
+const Navbar = () => {
+  const GITHUB_LOGO = '/icons/github.png';
+  const DEFAULT_AVATAR = '/images/Avatar.png';
+  const [userData, setUserData] = useState({});
+  const [toggle, setToggle] = useState(false);
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [mountedComponent, setMountedComponent] = useState(false);
+  const navbarRef = useRef();
+  GenericClosePopUp(navbarRef, () => {
+    setToggle(false);
+  });
+  useEffect(() => {
+    const fetchData = async () => {
+      await fetch(USER_DATA_URL, { credentials: 'include' })
+        .then((response) => {
+          if (!response.ok) {
+            setIsLoggedIn(false);
+            throw new Error(`${response.status} (${response.statusText})`);
+          }
+          return response.json();
+        })
+        .then((responseJson) => {
+          if (!responseJson.incompleteUserDetails) {
+            setIsLoggedIn(true);
+            setUserData({
+              userName: responseJson.username,
+              firstName: responseJson.first_name,
+            });
+          }
+          return window.location.replace('https://my.realdevsquad.com/signup');
+        })
+        .catch((err) => {
+          return err;
+        });
+      setMountedComponent(true);
+    };
+
+    fetchData();
+  }, []);
   return (
-    <div className={classNames.infoNavbar}>
-      <nav>
-        <div
-          className={classNames.DrawerToggle}
-          onClick={drawerToggleClicked}
-          onKeyPress={drawerToggleClicked}
-        >
-          <span />
-          <span />
-          <span />
-        </div>
-        <ul>
-          <li>
-            <a href="https://www.realdevsquad.com/">Home</a>
-          </li>
-          <li>
-            <a href="https://welcome.realdevsquad.com/">Welcome</a>
-          </li>
-          <li>
-            <a href="https://www.realdevsquad.com/events.html">Events</a>
-          </li>
-          <li>
-            <a
-              href="https://members.realdevsquad.com/"
-              style={{ color: '#87D870' }}
+    <div className={styles.wrapper}>
+      <>
+        <nav className={styles.navBar}>
+          <div
+            className={styles.hamburger}
+            ref={navbarRef}
+            onClick={() => setToggle(!toggle)}
+            onKeyDown={() => setToggle(!toggle)}
```

Comment:
Repetition! Let's make the anonymous function into a named function and reference it from here